server/handlers/handlers.go

```diff
@@ -101,25 +100,32 @@ func (ah *ActivityHandler) Start() {
 		}
 	}
 
+	// doQuit is called when the worker quits intentionally
+	doQuit := func() {
+		ah.workersWaitGroup.Done()
+	}
```

Comment:
This additional complexity helps me know when a particular worker is actually quitting. I contemplated putting this inside `isQuitting`, but the semantics felt weird since it's an API that is meant to be called repeatedly. Thoughts?